src/abi_impl.rs

```diff
@@ -98,3 +102,103 @@ pub(crate) fn assembly_script_print(env: &Env, arg: i32) {
         abi_bail!("Cannot read message pointer in memory");
     }
 }
+
+pub(crate) fn assembly_script_create_sc(env: &Env, bytecode: i32) -> i32 {
+    sub_remaining_point(env, settings::metering_create_sc());
+    let bytecode_ptr = StringPtr::new(bytecode as u32);
+    let memory = env.wasm_env.memory.get_ref().expect("uninitialized memory");
+    let address = if let Ok(bytecode) = &bytecode_ptr.read(memory) {
+        // Base64 to Binary
+        let bytecode = base64::decode(bytecode);
+        if bytecode.is_err() {
+            abi_bail!("Failed to decode module");
+        }
+        if let Ok(address) = create_sc(env, &bytecode.unwrap()) {
+            address
+        } else {
+            abi_bail!("Failed to create module smart contract");
+        }
+    } else {
+        abi_bail!("Cannot read bytecode pointer in memory");
+    };
+    if let Ok(address_ptr) = StringPtr::alloc(&address, &env.wasm_env) {
+        address_ptr.offset() as i32
+    } else {
+        abi_bail!("Cannot allocate address in memory")
+    }
+}
+
+pub(crate) fn assembly_script_set_data(env: &Env, key: i32, value: i32) {
+    sub_remaining_point(env, settings::metering_set_data());
```

Comment:
in a followup gas cost must depend on key and value size

Reply:
done!